smurfs/src/reducers/SmurfReducer.js

```diff
@@ -0,0 +1,32 @@
+import {
+    ADD_SMURF_SUCCESS,
+    ADD_SMURF_FAIL,
+    GET_SMURF_SUCCESS,
+    GET_SMURF_FAIL
+} from '../actions/index';
+
+export const initialState = {
+    villagers: []
+}
+
+export const smurfReducer = (state = initialState, action) => {
+    console.log("In smurfReducer", state);
+    switch(action.type) {
+        case ADD_SMURF_SUCCESS: {
+            console.log("Successfully added Smurf", state);
+            return {
+                ...state,
+                villagers: action.payload
```

Comment:
Spreading state here is redundant since villagers is the only state this reducer has access to.